### ISSUE TITLE
docs: add tensor_ext canonicalization documentation somewhere under the Dialect/TensorExt page

### ISSUE THREAD

**Issue by asraa** [google/heir#722]
On the website, we can either append it as a section on the single TensorExt page: https://heir.dev/docs/dialects/tensorext/, or we can add a sub-page about Design or more info?

**Comment by j2kun** [comment]
Canonicalization rules are weird in that, to my knowledge, there's no way to auto-generate docs from the tblgen. So this would have to be a manual process. I'll ask around in the MLIR world to see if there's a better way.

**Comment by asraa** [comment]
Right! My feeling was putting a manual description in a README.md as a section.

**Comment by j2kun** [comment]
Posted https://discourse.llvm.org/t/autogenerated-documentation-for-drr-patterns/79426
